pkg/node/node.go

```diff
@@ -53,20 +55,36 @@ func ListNodes(ctx context.Context, kubeClient client.Client, options *client.Li
 
 func EnsureNodePlugins(ctx context.Context, nodeObj *corev1.Node, kubeClient client.Client) error {
 	klog.InfoS("EnsureNodePlugins", "node", klog.KObj(nodeObj))
-	//does node have vhd installed
-	foundNvidiaPlugin, err := checkAndInstallNvidiaPlugin(ctx, nodeObj, kubeClient)
-	if err != nil {
-		return err
-	}
-	//does node have the custom label for DADI
-	foundDADIPlugin, err := checkAndInstallDADI(ctx, nodeObj, kubeClient)
-	if err != nil {
-		return err
-	}
-	if foundNvidiaPlugin && foundDADIPlugin {
-		return nil
+
+	var foundNvidiaPlugin, foundDADIPlugin bool
+	var err error
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			time.Sleep(1 * time.Second)
+			foundNvidiaPlugin, err = checkAndInstallNvidiaPlugin(ctx, nodeObj, kubeClient)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return err
+				} else {
+					continue
+				}
+			}
+			foundDADIPlugin, err = checkAndInstallDADI(ctx, nodeObj, kubeClient)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return err
+				} else {
+					continue
+				}
+			}
+			if foundDADIPlugin && foundNvidiaPlugin {
+				return nil
+			}
```

Comment:
You may move sleep in the end. The thought is that if indeed we need to sleep, we can update status indicating that we are waiting for addon to be ready.

Reply:
I had to move the func to the controller to be able to update the status. Done.